### PR TITLE
support older sbt 1.x versions

### PR DESCRIPTION
**Pull request by bjaglin**
https://github.com/thesamet/sbt-protoc/pull/183#issuecomment-716662257

### CODE REVIEW

**Comment by bjaglin** [comment]
Looking at the failure now

File: src/main/scala/sbtprotoc/ProtocPlugin.scala [hunk 1]
```diff
@@ -190,14 +188,17 @@ object ProtocPlugin extends AutoPlugin {
...
-                  tmpDir,
-                  log
-                ).fold(w => throw w.resolveException, _.head)
-              }
+              IO.withTemporaryDirectory(
+                { tmpDir =>
+                  lm.retrieve(
+                    moduleId,
+                    None,
+                    tmpDir,
```
Comment (bjaglin) [inline src/main/scala/sbtprotoc/ProtocPlugin.scala:196]: `retrieveDirectory` seems to be ignored when LM is backed by Coursier - a permanent coursier path is always returned. That's why tests are currently passing despite the current default `keepDirectory = false`.

**Comment by bjaglin** [comment]
Just found out what the problem was - a behavior change after sbt 1.2.x (at least when `useCoursier := true`).

I didn't see the problem locally as I guess I had a concurrent sbt 1.4.1 run copying the binary inside my user cache between 2 scripted invocations. Maybe scripted runs should set a custom, temporary `PROTOC_CACHE` since it's hard to troubleshoot otherwise?

**Comment by bjaglin** [comment]
Good job on https://github.com/thesamet/sbt-protoc/commit/56aa46c4a6906b0f55eab923a6e16e8d89650461, good to see some green!

File: src/main/scala/sbtprotoc/ProtocPlugin.scala (hunk 1 above)
Comment (thesamet) [reply src/main/scala/sbtprotoc/ProtocPlugin.scala:196]: Sounds like sbt 1.2.x actually respects the download directory which is automatically wiped before the downloaded executable is copied out of it? I would still like to have the temp directory automatically deleted. If my understanding is correct, the scope of `IO.withTemporaryDirectory` is incorrect.

Comment (bjaglin) [reply src/main/scala/sbtprotoc/ProtocPlugin.scala:196]: > Sounds like sbt 1.2.x actually respects the download directory which is automatically wiped before the downloaded executable is copied out of it?

Exactly. sbt 1.2.x and I believe later versions with coursier disabled.

>  I would still like to have the temp directory automatically deleted. If my understanding is correct, the scope of IO.withTemporaryDirectory is incorrect.

Looking at the current design of `FileCache`, I don't think that's possible. The copy operation could be exposed to the client by providing a callback?

**Comment by thesamet** [comment]
I fixed the bug that deletes the downloaded file before it's passed to the cache, so keepDirectory can remain false. Will you be able to rebase and test?

**Comment by bjaglin** [comment]
@thesamet done :+1: 

An unrelated addition/note: since setting `pluginCrossBuild / sbtVersion` indirectly sets [the default version for scripted execution](https://github.com/sbt/sbt/blob/1769a55fedbf221456ba584e6de9302ff647ad23/main/src/main/scala/sbt/ScriptedPlugin.scala#L61),  all scripted now run with that version sbt 1.2.8 (unless explicitly set in `build.properties`). This could be changed globally by defining `scriptedSbt`, but I guess it's fine - I just added a scripted in compat for each sbt minor version to detect any major regression specific to a minor version.

**Comment by thesamet** [comment]
Merged! Thanks for working on this. I think it would be nice to find a way to run the script tests for each minor version of SBT through github actions matrix. This way the whole suite will be tested.

**Pull request merged**